lib/edit-site-page.php

```diff
@@ -73,6 +74,12 @@ function gutenberg_edit_site_init( $hook ) {
 		$settings['fontSizes'] = $font_sizes;
 	}
 
+	// Get root template by trigerring `./template-loader.php`'s logic.
+	get_front_page_template();
+	get_index_template();
+	apply_filters( 'template_include', null );
```

Comment:
What is this line about? Should we comment?
I would have expected something like 

```
php
$template_id = get_front_page_template();
```

Reply:
We need to run the filters so that the `template-loader` code runs.

Reply:
How does this prevent us from having a dedicated function that returns the template?

Reply:
The way we hijack the template hierarchy is by filtering the template getter functions to store the template hierarchy in a global and then filtering `template_include` to return the template canvas with the found block template. There is no straightforward way to "return" the IDs from there.

This code is doing the same thing the front-end renderer does to set all those things in motion, but instead of echoing the template canvas at the end, it just uses the set globals to get the IDs it needs.